pennylane/transforms/hamiltonian_expval.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the hamiltonian expectation value tape transform
+"""
+import itertools
+import numpy as np
+import pennylane as qml
+
+
+def hamiltonian_expval(tape):
```

Comment:
I feel like `hamiltonian_expand` might be a more descriptive name here? Since, what the function does, is expand a given tape at the `expval(H)` point :)